### PR TITLE
refactor: Order validation with array

### PR DESCRIPTION
**Pull request by 0xJurassicPunk**

### CODE REVIEW

**Review by wjpg — DISMISSED** [review]

File: contracts/orderValidation/OrderValidatorV1.sol [hunk 1]
```diff
@@ -440,12 +447,12 @@ contract OrderValidatorV1 {
...
-     * @notice Check validity of ERC-1271 maker order
-     * @param digest digest
-     * @param targetSigner the signer address to confirm message validity
-     * @param v parameter (27 or 28)
-     * @param r parameter
-     * @param s parameter
+     * @notice Check the validity of ERC-1271 maker order
+     * @param digest Digest
+     * @param targetSigner Expect signer address to confirm message validity
+     * @param v Parameter (27 or 28)
```
Comment (0xhiroshi) [inline contracts/orderValidation/OrderValidatorV1.sol:453]: in L428-430 it was documented like

`v V parameter (27 or 28)`

but here

`v Parameter (27 or 28)`

File: test/helpers/order-validation-helper.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  makerOrder: MakerOrderWithSignature,
+  ERROR_CODE: number,
+  orderValidatorV1: Contract
+): Promise<void> {
+  const res = await orderValidatorV1.checkOrderValidity(makerOrder);
+  let arraySlot: number;
+  if (ERROR_CODE % 100 !== 0) {
+    arraySlot = Math.floor(ERROR_CODE / 100) - 1;
+  } else {
+    arraySlot = Math.floor(ERROR_CODE / 100);
```
Comment (0xhiroshi) [inline test/helpers/order-validation-helper.ts:16]: if ERROR_CODE % 100 === 0 then there shouldn't be a need to do Math.floor as it is an even number?

Comment (0xJurassicPunk) [reply test/helpers/order-validation-helper.ts]: Fixed!

File: contracts/orderValidation/OrderValidatorV1.sol (hunk 1 above)
Comment (0xJurassicPunk) [reply contracts/orderValidation/OrderValidatorV1.sol]: Adjusted!

**Pull request merged**